### ISSUE TITLE
就论文的复现有几个问题想请教一下

### ISSUE THREAD

**Issue by breezero1994** [MIVRC/MSRN-PyTorch#4]
作者 你好，这篇文章融合不同尺度的特征图来获取更好的结果，我觉得很受启发。所以我尝试着复现论文中的结果，下面这个是现在正在跑的结果。
epoch 17
Avg bicubic psnr = 33.691134694590936
Avg predicted psnr = 37.55205991379186
Avg bicubic ssim = 0.8475448663228888
Avg predicted ssim = 0.8952058433758587
1. 因为实验室的设备问题，DIV2K这样大的数据集跑不了，所以我仅仅将训练集换成了那291张图片，其他没做改变。所以我想问一下，这样的改变会对PSNR产生多大的影响，我测试了一下，图片x2的情况，epoch到9达到了极大值37.63，现在到了epoch17，徘徊在37.5左右。源代码是20epoch，我想问一下，你觉得还有必要跑下去吗？
2. 在做test的时候，SSIM明显是不对的，数据在上面。所以我想问是不是代码出了问题，还是我的设备出了问题。
希望得到作者的解答，谢谢！

**Comment by CV-JunchengLi** [comment]
您好，感谢对我们论文的关注。
我不确定你使用的是我们提供的旧代码还是新代码？
我们没有使用291作为训练集训练过网络，所以我不确定你们的结果是否正确。但根据我们的经验，数据集的质量和数量会很大程度影响模型最后的结果。对于一个参数较多的模型，训练集偏少会导致网络没有得到充分的训练。我们提供的旧代码中，使用MATLAB代码预处理DIV2K数据集并生成一个总的.h5文件。这个.h5文件高达几百个G，这对于你的设备来说可能负担是巨大的。因此，我们参照了EDSR开源的代码，提供了一份关于MSRN新的可运行代码，该份代码也采用DIV2K作为训练集，但是在数据处理和读取方面，EDSR的作者做了很好的处理，所以你无需很好的设备也可以正常运行该实验。我们按照EDSR提供的代码，重写了MSRN，并且重新训练该网络，得到的结果与我们ECCV论文上的结果是接近的。在3、4倍上甚至达到了更好的结果。与旧代码不同的是，新代码与近期SR的工作一样，都是直接在RGB3通道上进行训练的。而在测试阶段，我们只测试了图像在Y通道上的PSNR和SSIM。

**Comment by userzm** [comment]
作者你好 新版本的代码里的trainer.py中的28行左右for _ in range(len(ckp.log)): self.scheduler.step()这行是不是有问题，学习率好像不是中断训练之前的那个学习率了，继续训练的学习率变得更小了
希望得到作者的解答，谢谢！

**Comment by breezero1994** [comment]
您好，我不是原文的作者，另外我复现论文的时候是在old版本上进行的。新版本是在EDSR的主框架上修改的，在old的里面，学习率就有衰减，所以学习率就是会变小的。



---原始邮件---
发件人:&nbsp;"Areil"<notifications@github.com&gt;
发送时间:&nbsp;2019年10月15日(星期二) 晚上7:06
收件人:&nbsp;"MIVRC/MSRN-PyTorch"<MSRN-PyTorch@noreply.github.com&gt;;
抄送:&nbsp;"Author"<author@noreply.github.com&gt;;"breezero1994"<1109080397@qq.com&gt;;
主题:&nbsp;Re: [MIVRC/MSRN-PyTorch] 就论文的复现有几个问题想请教一下 (#4)



作者你好 新版本的代码里的trainer.py中的28行左右for _ in range(len(ckp.log)): self.scheduler.step()这行是不是有问题，学习率好像不是中断训练之前的那个学习率了，继续训练的学习率变得更小了
希望得到作者的解答，谢谢！

**Comment by CV-JunchengLi** [comment]
不好意思，没看懂你这句话想要表达什么“学习率好像不是中断训练之前的那个学习率了，继续训练的学习率变得更小了”

**Comment by userzm** [comment]
就是这里面https://blog.csdn.net/sinat_31987445/article/details/94736301提到的第二个问题，他提到学习率异常，但是我测试了下学习率好像没问题，所以就来问问作者

**Comment by CV-JunchengLi** [comment]
每次运行的时候都会打印出学习率，这个学习率是真实的，你看看每次打印出来的学习率是否正确就是。我看了你提供链接的博士，他在更高版本的pytorch上运行这份代码，pytorch不向下兼容，可能会导致一系列问题。

**Comment by CV-JunchengLi** [comment]
每次运行的时候都会打印出学习率，这个学习率是真实的，你看看每次打印出来的学习率是否正确就行。我看了你提供链接的博主，他在更高版本的pytorch上运行这份代码，pytorch不向下兼容，可能会导致一系列问题。
